File: fixtures/scale/ruby/large/lib/check_weavehash.rb
require_relative 'weavehash_digest'
require_relative 'weavehash_digest_pair'
require_relative 'corpus_hash'

def check_weavehash
  corpus_hash.each do |text|
    first = weavehash_digest(text)
    second = weavehash_digest(text)
    return false if first != second || first.length != 8
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_weavehash.py
from cipherlab.ciphers.weavehash import weavehash_digest
from cipherlab.specs.weavehash_spec import weavehash_spec
from cipherlab.corpus.corpus_hash import corpus_hash


def check_weavehash():
    spec = weavehash_spec()
    for text in corpus_hash():
        first = weavehash_digest(text)
        second = weavehash_digest(text)
        if first != second or len(first) != 8:
            return False
    return spec["category"] == "hash"
